core/src/test/scala/integration/kafka/api/KafkaAdminClientIntegrationTest.scala

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.api
+
+import java.util
+import java.util.Properties
+import java.util.concurrent.ExecutionException
+
+import kafka.integration.KafkaServerTestHarness
+import kafka.server.KafkaConfig
+import org.apache.kafka.clients.admin._
+import kafka.utils.{Logging, TestUtils}
+import org.apache.kafka.clients.admin.NewTopic
+import org.apache.kafka.common.KafkaFuture
+import org.apache.kafka.common.errors.TopicExistsException
+import org.apache.kafka.common.protocol.ApiKeys
+import org.junit.{Before, Rule, Test}
+import org.junit.rules.Timeout;
+import org.junit.Assert._
+
+import scala.collection.JavaConverters._
+
+/**
+ * An integration test of the KafkaAdminClient.
+ *
+ * Also see KafkaAdminClientUnitTest for a unit test of the admin client.
+ */
+class KafkaAdminClientIntegrationTest extends KafkaServerTestHarness with Logging {
```

Comment:
It would be good to run these tests with security enabled. Maybe a variant with SASL_SSL would do the job. It's reasonably straightforward, see `SaslSslConsumerTest`.